src/wasmtime/core.rs

```diff
@@ -31,7 +32,23 @@ impl WasmValue for wasmtime::Val {
     type Type = wasmtime::ValType;
 
     fn ty(&self) -> Self::Type {
-        self.ty()
+        let config = wasmtime::Config::new();
+        let engine = wasmtime::Engine::new(&config).unwrap();
+        let store = wasmtime::Store::new(&engine, ());
+        self.ty(store)
```

Comment:
I think this can be handled with a `match self`. There are `const`s for ref types (which I haven't really wrapped my head around yet), e.g. https://docs.rs/wasmtime/20.0.2/wasmtime/struct.RefType.html#associatedconstant.EXTERNREF

Reply:
Nice. At least the names match up and they are constants. See the next commit.